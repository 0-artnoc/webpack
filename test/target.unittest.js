const { resolve } = require("../lib/config/browserslistTargetHandler");

describe("browserslist target", () => {
	const tests = [
		["ie 11"],
		["chrome 80"],
		["node 0.10"],
		["node 0.12"],
		["node 10"],
		["node 10.0"],
		["node 10.17"],
		["node 10.0.0"],
		["node 10.17.0"],
		["node 12"],
		["safari 10"],
		["safari TP"],
		["safari 11"],
		["safari 12.4"],
		["firefox 80 ", "chrome 80"]
	];

	for (const test of tests) {
		it(`${JSON.stringify(test)}`, () => {
			expect(resolve(test)).toMatchSnapshot();
		});
	}
});
