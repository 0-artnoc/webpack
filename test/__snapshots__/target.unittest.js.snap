// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`browserslist target ["chrome 80"] 1`] = `
Object {
  "arrowFunction": true,
  "bigIntLiteral": true,
  "browser": true,
  "const": true,
  "destructuring": true,
  "document": true,
  "dynamicImport": true,
  "dynamicImportInWorker": true,
  "electron": false,
  "fetchWasm": true,
  "forOf": true,
  "global": false,
  "globalThis": true,
  "importScripts": false,
  "importScriptsInWorker": true,
  "module": true,
  "node": false,
  "nodeBuiltins": false,
  "nwjs": false,
  "require": false,
  "web": true,
  "webworker": false,
}
`;

exports[`browserslist target ["chrome 80.0"] 1`] = `
Object {
  "arrowFunction": false,
  "bigIntLiteral": false,
  "browser": true,
  "const": false,
  "destructuring": true,
  "document": true,
  "dynamicImport": false,
  "dynamicImportInWorker": false,
  "electron": false,
  "fetchWasm": true,
  "forOf": true,
  "global": false,
  "globalThis": true,
  "importScripts": false,
  "importScriptsInWorker": true,
  "module": false,
  "node": false,
  "nodeBuiltins": false,
  "nwjs": false,
  "require": false,
  "web": true,
  "webworker": false,
}
`;

exports[`browserslist target ["firefox 80 ","chrome 80"] 1`] = `
Object {
  "arrowFunction": false,
  "bigIntLiteral": false,
  "browser": true,
  "const": false,
  "destructuring": true,
  "document": true,
  "dynamicImport": false,
  "dynamicImportInWorker": false,
  "electron": false,
  "fetchWasm": true,
  "forOf": true,
  "global": false,
  "globalThis": true,
  "importScripts": false,
  "importScriptsInWorker": true,
  "module": false,
  "node": false,
  "nodeBuiltins": false,
  "nwjs": false,
  "require": false,
  "web": true,
  "webworker": false,
}
`;

exports[`browserslist target ["ie 11"] 1`] = `
Object {
  "arrowFunction": false,
  "bigIntLiteral": false,
  "browser": true,
  "const": true,
  "destructuring": false,
  "document": true,
  "dynamicImport": false,
  "dynamicImportInWorker": false,
  "electron": false,
  "fetchWasm": true,
  "forOf": false,
  "global": false,
  "globalThis": false,
  "importScripts": false,
  "importScriptsInWorker": true,
  "module": false,
  "node": false,
  "nodeBuiltins": false,
  "nwjs": false,
  "require": false,
  "web": true,
  "webworker": false,
}
`;

exports[`browserslist target ["node 6"] 1`] = `
Object {
  "arrowFunction": false,
  "bigIntLiteral": false,
  "browser": false,
  "const": false,
  "destructuring": true,
  "document": false,
  "dynamicImport": false,
  "dynamicImportInWorker": false,
  "electron": false,
  "fetchWasm": false,
  "forOf": true,
  "global": true,
  "globalThis": false,
  "importScripts": false,
  "importScriptsInWorker": true,
  "module": false,
  "node": true,
  "nodeBuiltins": true,
  "nwjs": false,
  "require": true,
  "web": false,
  "webworker": false,
}
`;

exports[`browserslist target ["node 6.0"] 1`] = `
Object {
  "arrowFunction": false,
  "bigIntLiteral": false,
  "browser": false,
  "const": false,
  "destructuring": true,
  "document": false,
  "dynamicImport": false,
  "dynamicImportInWorker": false,
  "electron": false,
  "fetchWasm": false,
  "forOf": true,
  "global": true,
  "globalThis": false,
  "importScripts": false,
  "importScriptsInWorker": true,
  "module": false,
  "node": true,
  "nodeBuiltins": true,
  "nwjs": false,
  "require": true,
  "web": false,
  "webworker": false,
}
`;

exports[`browserslist target ["node 6.0.0"] 1`] = `
Object {
  "arrowFunction": false,
  "bigIntLiteral": false,
  "browser": false,
  "const": false,
  "destructuring": true,
  "document": false,
  "dynamicImport": false,
  "dynamicImportInWorker": false,
  "electron": false,
  "fetchWasm": false,
  "forOf": true,
  "global": true,
  "globalThis": false,
  "importScripts": false,
  "importScriptsInWorker": true,
  "module": false,
  "node": true,
  "nodeBuiltins": true,
  "nwjs": false,
  "require": true,
  "web": false,
  "webworker": false,
}
`;
